### ISSUE TITLE
Tailwind fails with styled-jsx

### ISSUE THREAD

**Issue by DaniGuardiola** [vercel/next.js#19477]
# Bug report

## Describe the bug

I don't know if I'm doing something wrong, but I can't get Tailwind to work with styled-jsx.

Relevant issue from Tailwind repo (closed as it seems to be a Next.js problem): https://github.com/tailwindlabs/tailwindcss/issues/2870#issuecomment-731811135

Also, it could be useful to add this configuration to the with-tailwind example. I could send a PR for that once I get it working.

Thank you! :)

## To Reproduce

Here's a minimal reproduction: [DaniGuardiola/tmp-tailwind-bug-repro](https://github.com/DaniGuardiola/tmp-tailwind-bug-repro)

## Expected behavior

PostCSS should be used for styled-jsx styles, including Tailwind stuff.

## Actual behavior

I get this result:

```bash
wait  - compiling...
[styled-jsx] Loading plugin from path: /home/dani/projects/daniguardiola-me/node_modules/styled-jsx-plugin-postcss/index.js
Next.js 9 default postcss support uses a non standard postcss config schema https://err.sh/next.js/postcss-shape, you must use the interoperable object-based format instead https://nextjs.org/docs/advanced-features/customizing-postcss-config
Next.js 9 default postcss support uses a non standard postcss config schema https://err.sh/next.js/postcss-shape, you must use the interoperable object-based format instead https://nextjs.org/docs/advanced-features/customizing-postcss-config
error - ./components/Post.tsx
Error: /home/dani/projects/daniguardiola-me/components/Post.tsx: postcss failed with TypeError: Invalid PostCSS Plugin found: [0]
    at Array.forEach (<anonymous>)
```

Although [Tailwind's maintainer doesn't get this error](https://github.com/tailwindlabs/tailwindcss/issues/2870#issuecomment-731781917), but Tailwind still doesn't work in the case.

## System information

- OS: Linux (Pop! OS 20.10)
- Version of Next.js: 10.0.2
- Version of Node.js: v14.7.0
- Deployment: next dev

**Comment by timneutkens** [comment]
You're using a community package together with styled-jsx which depends on PostCSS 7, tailwind by default support postcss 8, so you'll want to open an issue on the community project: https://github.com/giuseppeg/styled-jsx-plugin-postcss/blob/master/package.json#L25

**Comment by DaniGuardiola** [comment]
Thanks @timneutkens.

For future visitors of this issue, here's the relevant issue in that plugin: https://github.com/giuseppeg/styled-jsx-plugin-postcss/issues/42

Also @timneutkens just wanted your input on something. I'm using MDX and I want to style the rendered markdown using tailwind. That's why I wanted to make use of the scoping that styled-jsx provides. I know I can probably configure the providers and use custom elements substituting, for example, all <a> or <p> elements, but still I wonder why is that recommendation? What is the difference between using tailwind (`@apply`) in one vs. the other?

Thank you for your time :)

**Comment by DaniGuardiola** [comment]
This issue has been fixed yay! Just update to 4.0

https://github.com/giuseppeg/styled-jsx-plugin-postcss/issues/42

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
